graphz/src/main/scala/coop/rchain/graphz/Graphz.scala

```diff
@@ -1,104 +1,104 @@
+class ListSerializer[F[_]: Concurrent](ref: Ref[F, Vector[String]]) extends GraphSerializer[F] {
+  def push(str: String, suffix: String): F[Unit] = ref.update { current =>
+    current :+ (str + suffix)
+  }
+  def show: F[String] = ref.get.map { _.mkString("") }
 }
 
 class FileSerializer[F[_]: Sync](fos: FileOutputStream) extends GraphSerializer[F] {
   def push(str: String, suffix: String): F[Unit] = Sync[F].delay {
     fos.write(str.getBytes)
     fos.flush()
   }
+  override def show: F[String] = ???
 }
 
 sealed trait GraphType
-final case object Graph   extends GraphType
-final case object DiGraph extends GraphType
+case object Graph   extends GraphType
+case object DiGraph extends GraphType
 
 sealed trait GraphShape
-final case object Circle       extends GraphShape
-final case object DoubleCircle extends GraphShape
-final case object Box          extends GraphShape
-final case object PlainText    extends GraphShape
-final case object Msquare      extends GraphShape
-final case object Record       extends GraphShape
+case object Circle       extends GraphShape
+case object DoubleCircle extends GraphShape
+case object Box          extends GraphShape
+case object PlainText    extends GraphShape
+case object Msquare      extends GraphShape
+case object Record       extends GraphShape
 
 sealed trait GraphRank
-final case object Same   extends GraphRank
-final case object Min    extends GraphRank
-final case object Source extends GraphRank
-final case object Max    extends GraphRank
-final case object Sink   extends GraphRank
+case object Same   extends GraphRank
+case object Min    extends GraphRank
+case object Source extends GraphRank
+case object Max    extends GraphRank
+case object Sink   extends GraphRank
 
 sealed trait GraphRankDir
-final case object TB extends GraphRankDir
-final case object BT extends GraphRankDir
-final case object LR extends GraphRankDir
-final case object RL extends GraphRankDir
+case object TB extends GraphRankDir
+case object BT extends GraphRankDir
+case object LR extends GraphRankDir
+case object RL extends GraphRankDir
 
 sealed trait GraphStyle
-final case object Solid  extends GraphStyle
-final case object Bold   extends GraphStyle
-final case object Filled extends GraphStyle
-final case object Invis  extends GraphStyle
-final case object Dotted extends GraphStyle
-final case object Dashed extends GraphStyle
+case object Solid  extends GraphStyle
+case object Bold   extends GraphStyle
+case object Filled extends GraphStyle
+case object Invis  extends GraphStyle
+case object Dotted extends GraphStyle
+case object Dashed extends GraphStyle
 
 sealed trait GraphArrowType
-final case object NormalArrow extends GraphArrowType
-final case object InvArrow    extends GraphArrowType
-final case object NoneArrow   extends GraphArrowType
+case object NormalArrow extends GraphArrowType
+case object InvArrow    extends GraphArrowType
+case object NoneArrow   extends GraphArrowType
```

Comment:
What to remove _final_ for case classes?

Reply:
Reverted in [commit](https://github.com/rchain/rchain/pull/3574/commits/a175818472f857fbeb97f59e15ce527b7c89ad92). It was IDE warning `'final' modifier is redundant for toplevel objects`

Reply:
It seems that `final` modifier is added by default for all `object` definitions and possibly forbidden in the future versions.
https://github.com/scala/bug/issues/11094

I'm OK that we are not using final for objects and remove it on existing code.

It makes definitions with objects and classes little strange but this is Scala. :)
```scala
sealed trait A
case object B extends A
final class C(a: Int) extends A
```

Reply:
`final` keyword was removed in [commit](https://github.com/rchain/rchain/pull/3574/commits/dc1aff6d36a7fe2140b1795dbd8d0ca256fc68f2) to make the code clear.